qa/integration-tests/src/test/java/io/camunda/zeebe/it/gateway/GatewayAuthenticationIdentityIT.java

```diff
@@ -0,0 +1,296 @@
+                assertThat(statusRuntimeException.getStatus())
+                    .hasFieldOrPropertyWithValue("code", Status.UNAUTHENTICATED.getCode())
+                    .hasFieldOrPropertyWithValue(
+                        "description",
+                        "Expected bearer token at header with key [authorization], but found nothing");
+              });
+    }
+  }
+
+  @Test
+  void deployModelFailsWithInvalidAuthToken() {
+    // given
+    try (final var client =
+        cluster
+            .newClientBuilder()
+            .defaultRequestTimeout(Duration.ofMinutes(1))
+            .credentialsProvider(new InvalidAuthTokenProvider())
+            .build()) {
+      // when
+      final var deploymentFuture =
+          client
+              .newDeployResourceCommand()
+              .addProcessModel(PROCESS_MODEL, "model.bpmn")
+              .send()
+              .toCompletableFuture();
+
+      // then
+      assertThat(deploymentFuture)
+          .failsWithin(Duration.ofSeconds(1))
+          .withThrowableOfType(ExecutionException.class)
+          .withCauseInstanceOf(StatusRuntimeException.class)
+          .extracting(
+              Throwable::getCause, as(InstanceOfAssertFactories.type(StatusRuntimeException.class)))
+          .satisfies(
+              statusRuntimeException -> {
+                assertThat(statusRuntimeException.getStatus())
+                    .hasFieldOrPropertyWithValue("code", Status.UNAUTHENTICATED.getCode())
+                    .hasFieldOrPropertyWithValue(
+                        "description", "Failed to parse bearer token, see cause for details");
+              });
+    }
+  }
+
+  @Test
+  void deployModelSucceedsWithValidAuthToken() {
+    // given
+    awaitCamundaRealmAvailabilityOnKeycloak();
+
+    try (final var client =
+        cluster
+            .newClientBuilder()
+            .defaultRequestTimeout(Duration.ofMinutes(1))
+            .credentialsProvider(
+                new OAuthCredentialsProviderBuilder()
+                    .clientId(ZEEBE_CLIENT_ID)
+                    .clientSecret(ZEEBE_CLIENT_SECRET)
+                    .audience(ZEEBE_CLIENT_AUDIENCE)
+                    .authorizationServerUrl(
+                        getKeycloakRealmAddress() + "/protocol/openid-connect/token")
+                    .build())
+            .build()) {
+      // when
+      final var deploymentFuture =
+          client
+              .newDeployResourceCommand()
+              .addProcessModel(PROCESS_MODEL, "model.bpmn")
+              .send()
+              .toCompletableFuture();
+
+      // then
+      assertThat(deploymentFuture).succeedsWithin(Duration.ofSeconds(1));
+    }
+  }
+
+  /**
+   * Awaits the presence of the Camunda realm on the keycloak container. Once Keycloak and Identity
+   * booted up, Identity will eventually configure the Camunda Realm on Keycloak.
+   */
+  private static void awaitCamundaRealmAvailabilityOnKeycloak() {
```

Comment:
how is identity waiting for this?

Reply:
@megglos we do not have any internal wait mechanism. Looking at testcontainer code the default wait strategy is wait until there is a successful response on the configured port.

Reply:
BTW: we moved away from KeycloakContainer to using a generic testcontainer, see https://github.com/camunda-cloud/identity/blob/main/management-api/src/test/java/io/camunda/identity/KeycloakBaseIT.java#L75

I think the reason was that at that time KeycloakContainer did not support v19. On the other hand using the generic container requires starting another container for postgres. So if your solution works, then it's fine :)

Reply:
Good job we're looking at supporting version 20! :) - another reason for the setup we have in our ITs is because this also aligns with the distribution/setup suggested in our Helm charts